lib/libxdp/libxdp.c

```diff
@@ -2157,11 +2235,11 @@ int xdp_multiprog__detach(struct xdp_multiprog *mp)
 	if (!mp || !mp->is_loaded)
 		return -EINVAL;
 
-	err = xdp_multiprog__attach(mp, NULL, mp->attach_mode);
+	err = xdp_multiprog__attach(mp, NULL, XDP_MODE_UNSPEC);
```

Comment:
Eh? Why? :)

Reply:
Since xdp_multiprog__attach detaches only the specified mode, and now we can have two modes at once, I've changed it so that XDP_MODE_UNSPEC detaches every mode. Since xdp_multiprog__detach detaches entire multiprog.

Reply:
Right, but see the comment about just not handling hw mode in xdp_multiprog__attach() at all :)